### PR TITLE
Support mock time and ticker

### PR DESCRIPTION
**Pull request by luckyxiaoqiang**
### Describe what this PR does / why we need it
Support mock time and ticker. We can use mock time and ticker to do unit test, which can decouple from system real time. 

### Special notes for reviews

### CODE REVIEW

File: core/flow/slot.go [hunk 1]
```diff
@@ -44,6 +43,8 @@ func (s *Slot) Order() uint32 {
 }
 
 func (s *Slot) Check(ctx *base.EntryContext) *base.TokenResult {
+	util.SetClock(util.NewMockClock())
```
Comment (sczyh30) [inline core/flow/slot.go:46]: Is it needed here?

Comment (luckyxiaoqiang) [reply core/flow/slot.go:46]: > Is it needed here?

That's a mistake.

File: util/time.go [hunk 2]
```diff
@@ -24,6 +26,261 @@ const (
...
+
+	_ Ticker = &RealTicker{}
+	_ Ticker = &MockTicker{}
+
+	_ TickerCreator = &RealTickerCreator{}
+	_ TickerCreator = &MockTickerCreator{}
+)
+
+var (
+	realClock    *RealClock
```
Comment (louyuting) [inline util/time.go:41]: Maybe this field is unnecessary?

File: util/time.go [hunk 3]
```diff
@@ -24,6 +26,261 @@ const (
...
+	currentClock *atomic.Value
+
+	realTickerCreator    *RealTickerCreator
```
Comment (louyuting) [inline util/time.go:44]: Maybe this field is unnecessary?

File: util/time.go [hunk 4]
```diff
@@ -36,15 +293,21 @@ func FormatDate(tsMillis uint64) string {
...
 }
 
 // Returns the current Unix timestamp in nanoseconds.
 func CurrentTimeNano() uint64 {
-	return uint64(time.Now().UnixNano())
+	return CurrentClock().CurrentTimeNano()
+}
+
+// Now returns the current local time.
+func Now() time.Time {
```
Comment (louyuting) [inline util/time.go:305]: Maybe we need a benchmark to test the performance diff between `Now()` and `CurrentTimeMillsWithTicker`

File: util/time.go (hunk 2 above)
Comment (luckyxiaoqiang) [reply util/time.go:41]: Now it is used to do a safe fallback like in `CurrentClock`.

File: util/time.go (hunk 3 above)
Comment (luckyxiaoqiang) [reply util/time.go:44]: Now it is used to do a safe fallback like in `CurrentTickerCreator`.

File: util/time.go (hunk 2 above)
Comment (luckyxiaoqiang) [reply util/time.go]: Refined.

File: util/time.go (hunk 4 above)
Comment (luckyxiaoqiang) [reply util/time.go:302]: Added. 
```
BenchmarkCurrentTimeInMs/CurrentTimeMillis-4            230781794                5.26 ns/op
BenchmarkCurrentTimeInMs/CurrentTimeMillisDirect-4      519137175                2.32 ns/op
BenchmarkNow/util.Now-4                                 14527578                83.3 ns/op             
BenchmarkNow/time.Now-4                                 14849454                77.0 ns/op             
```

**Review by louyuting — APPROVED** [review]
LGTM

**Review by sczyh30 — APPROVED** [review]
LGTM

**Pull request merged**